cmake/modules/Findcppzmq.cmake

```diff
@@ -8,8 +8,25 @@ find_path(cppzmq_INCLUDE_DIRS "zmq.hpp"
         )
 mark_as_advanced(cppzmq_INCLUDE_DIRS)
 
+# check for required minimum version for RooFitZMQ
+if(cppzmq_INCLUDE_DIRS)
+    SET(SAVE_CMAKE_REQUIRED_INCLUDES "${CMAKE_REQUIRED_INCLUDES}")
+    SET(CMAKE_REQUIRED_INCLUDES ${cppzmq_INCLUDE_DIRS} ${ZeroMQ_INCLUDE_DIRS})
+    SET(SAVE_CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES}")
+    SET(CMAKE_REQUIRED_LIBRARIES libzmq)
+    check_cxx_source_runs("#include <zmq.hpp> \n int main () {if (CPPZMQ_VERSION_MAJOR > 4 || (CPPZMQ_VERSION_MAJOR == 4 && CPPZMQ_VERSION_MINOR >= 8)) {return 0;} else {return 1;} }" cppzmq_VERSION_COMPATIBLE)
```

Comment:
The `Findcppzmq` module should not hard-code checks like this, you should instead detect which version is installed (see `FindLZ4.cmake` for an example) and export it as variables. Then you can call `find_package(cppzmq 4.8)` (i.e., with a version) to reject earlier versions.

Reply:
Actually, this module should be removed and ROOT should rely on what's provided by `cppzmq` itself.

Reply:
Are they also built using autotools? Why are the CMake modules not distributed? But in any case, even if using the external modules is impossible, the module in ROOT should still do a proper check for the version and export it as CMake variables so that `find_package` works with a minimum version as a parameter.

Reply:
Yep, they maintain both an autotools and a CMake build system. Most Linux distros use the autotools one.

A version check for `libzmq` should be included, but only after the new `libzmq` release. Right now, it will not be used, since we need a development build anyway.

For `cppzmq`, this PR adds the version check, admittedly in the most minimal/inflexible way possible, just meant to solve the bug that @Axel-Naumann found. I unfortunately do not have time to change the implementation here right now, so perhaps we can defer this upgrade to an issue for later.